src/main/java/com/onaio/steps/handler/actions/TakeSurveyHandler.java

```diff
@@ -113,7 +113,8 @@ public boolean canHandleResult(int requestCode) {
         }
     }
 
-    private String getValue(String key) {
-        return KeyValueStoreFactory.instance(activity).getString(key);
+    @Override
+    public void tryToResolve() {
```

Comment:
Do we need this abstraction?

Reply:
if we leave the `ODK collect` in recent apps for some time and try to fetch the form from the `Steps` app then might be a chance that the `Steps` app failed to get the form. To resolve this issue ` Steps` app tries to fetch once again.